src/panels/KaitoManagePanel.ts

```diff
@@ -0,0 +1,207 @@
+                const tempFilePath = join(tmpdir(), `kaito-deployment-${Date.now()}.yaml`);
+                writeFileSync(tempFilePath, modelYaml, "utf8");
+                const command = `apply -f ${tempFilePath}`;
+                const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+                if (failed(kubectlresult)) {
+                    vscode.window.showErrorMessage(`Error deploying 'workspace-${modelName}': ${kubectlresult.error}`);
+                    return;
+                }
+            });
+            vscode.window.showInformationMessage(`'workspace-${modelName}' has been redeployed.`);
+            await this.updateModels(webview);
+        } finally {
+            this.operatingState[modelName] = false;
+        }
+    }
+
+    // Updates the current state of models on the cluster
+    private async updateModels(webview: MessageSink<ToWebViewMsgDef>) {
+        const command = `get workspace -o json`;
+        const kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+        if (failed(kubectlresult)) {
+            webview.postMonitorUpdate({
+                clusterName: this.clusterName,
+                models: [],
+            });
+            return;
+        }
+
+        const models = [];
+        const data = JSON.parse(kubectlresult.result.stdout);
+        for (const item of data.items) {
+            const conditions: Array<{ type: string; status: string }> = item.status?.conditions || [];
+            const { resourceReady, inferenceReady, workspaceReady } = getConditions(conditions);
+            models.push({
+                name: item.inference?.preset?.name,
+                instance: item.resource?.instanceType,
+                resourceReady: resourceReady,
+                inferenceReady: inferenceReady,
+                workspaceReady: workspaceReady,
+                age: convertAgeToMinutes(item.metadata?.creationTimestamp),
+            });
+        }
+
+        webview.postMonitorUpdate({
+            clusterName: this.clusterName,
+            models: models,
+        });
+    }
+
+    private async handleMonitorUpdateRequest(webview: MessageSink<ToWebViewMsgDef>) {
+        await this.updateModels(webview);
+    }
+
+    // Retrieves the logs from the Kaito workspace and outputs them in a new text editor.
+    private async handleGetLogsRequest() {
+        await longRunning(`Retrieving logs`, async () => {
+            let command = `get po -l app=ai-toolchain-operator -A -o json`;
+            let kubectlresult = await invokeKubectlCommand(this.kubectl, this.kubeConfigFilePath, command);
+            if (failed(kubectlresult)) {
+                vscode.window.showErrorMessage(`Error fetching logs: ${kubectlresult.error}`);
+                return;
+            }
+            const data = JSON.parse(kubectlresult.result.stdout);
+            const items = data.items;
```

Comment:
💡 Suggestion only, please.

```*.ts
const items = data.items;
let latestPod = null;

items.forEach(item => {
    const { name, creationTimestamp } = item.metadata;
    
    if (name.startsWith("kaito-workspace")) {
        const creationDate = new Date(creationTimestamp);

        if (!latestPod || creationDate > latestPod.date) {
            latestPod = { name, date: creationDate };
        }
    }
});

if (!latestPod) {
    vscode.window.showErrorMessage("Error finding workspace pod.");
    return;
}

```

Reply:
This was fixed in a later PR, will make sure to merge those adjustments in here 👍